modules/account-lib/src/coin/dot/keyPair.ts

```diff
@@ -12,18 +19,56 @@ export class KeyPair extends Ed25519KeyPair {
     super(source);
   }
 
+  /**
+   * Helper function to create the KeyringPair for signing a dot transaction.
+   *
+   * @returns {KeyringPair} dot KeyringPair
+   * @see https://polkadot.js.org/docs/api/start/keyring
+   */
+  protected createPolkadotPair(): KeyringPair {
+    const secretKey = this.keyPair.prv ? new Uint8Array(Buffer.from(this.keyPair.prv, 'hex')) : new Uint8Array();
+    const publicKey = new Uint8Array(Buffer.from(this.keyPair.pub, 'hex'));
+    return createPair({ toSS58: keyring.encodeAddress, type: TYPE }, { secretKey, publicKey });
+  }
+
+  /** @inheritdoc */
   getAddress(format?: AddressFormat): string {
-    throw new Error('Method not implemented.');
+    return this.createPolkadotPair().address;
   }
 
+  /**
+   * Getting the KeyringPair for signing a dot transaction.
+   *
+   * @returns {KeyringPair} dot KeyringPair
+   */
+  getSigningKeyPair(): KeyringPair {
```

Comment:
How is this different than `getKeys`?

Reply:
getKeys returns pub and prv keys, this function returns the @polkadot library keyring instance which is used to sign the transactions we build using `@substrate/txwrapper-polkadot`

Reply:
This should be done in the `getKeys()`

Reply:
We should not expose lib specific objects, the whole point of account lib is to provide an abstraction layer around them. If we need the keys we should use `getKeys()`